joebvp/VPmeasure.py

```diff
@@ -606,15 +600,38 @@ def batch_fit(spec,filelist,outparfile=None,outmodelfile=None,**kwargs):
             q_pass += 1
         except:
             print('VPmeasure: Fitting failed:',ff)
+            fails += [ff]
             q_fail += 1
     print("")
     print("VPmeasure: {}/{} fits converged, {}/{} failed (see log for details).\n".format(q_pass, i, q_fail, i))
 
+    # ask whether run GUI mode on failures
+    if q_fail >= 0:
+        print("VPmeasure: Failed runs are: {}\n".format(fails))
+        while True:
+            answer = raw_input("VPmeasure: Would you like to run the failed fits in GUI mode? (y/n): ")
+            if answer in ['y','n', 'yes','no']:
+                break
+        if answer in ['y', 'yes']:
+            for ff in fails:
+                go(spec, ff)
+
+    # ask whether to concatenate and create fits inspection files
+    while True:
+        answer = raw_input("VPmeasure: Would you like to concatenate all .VP outputs in the working directory and create a FitsInspection.pdf file? (y/n): ")
+        if answer in ['y','n', 'yes','no']:
+            break
+    if answer in ['y', 'yes']:
+        concatenate_all()
+    else:
+        print("VPmeasure: Done.")
```

Comment:
I definitely like being prompted about whether to fit the failures in the GUI, but I don't like having to tell the thing that we want a concatenated file and inspection pdf.  For example, I may start a batch fit in the office on Friday afternoon, run them all night, and want to inspect them from home on Saturday morning.  In that case, it's very convenient to open up the fit inspection pdf and take a look.  Might I suggest we just include an optional argument to the function to specify if we *do not* want a concatenated parameter file and inspection pdf?

Reply:
Roger. Will avoid asking about this. The only issue remaining is that, without a proper way to close the GUI (i.e. without crashing it), whatever lines come next after the launch of the GUI will not be executed because the program crashes...